headerforwarder/forwarder.go

```diff
@@ -0,0 +1,123 @@
+package headerforwarder
+
+import (
+	"fmt"
+	"net/http"
+)
+
+// HeaderExtractingTransport is a utility to help a rosetta server forward headers to and from
+// native node requests. It implements several interfaces to achieve that:
+//   - http.RoundTripper: this can be used to create an http Client that will automatically save headers
+//     if necessary
+//
+// the headers can be requested later.
+//
+// TODO: this needs to expire entries after a certain amount of time
```

Comment:
Instead of expiring after a certain amount of time, can we delete an entry from requestHeaders after it is looked up and added to the final rosetta response's headers?

Reply:
This is a great idea (and one I meant to add but forgot). 

Eventually we should probably do both just because headers that are never accessed would stick around forever, but deleting entries as we use them will mean we can release this without expiration probably